benchmarks/benchmarks/bench_function_base.py

```diff
@@ -98,44 +98,122 @@ def time_select_larger(self):
-        np.random.seed(25)
-        np.random.shuffle(self.o)
-        # quicksort implementations can have issues with equal elements
-        self.equal = np.ones(10000, dtype=dtype)
-        self.many_equal = np.sort(np.arange(10000) % 10).astype(dtype)
+    ARRAY_SIZE = 10000
 
-        try:
-            np.sort(self.e, kind=kind)
-        except TypeError:
-            raise NotImplementedError()
+    def setup(self, kind, dtype, array_type):
+        np.random.seed(1234)
+        self.arr = getattr(self, 'array_' + array_type)(self.ARRAY_SIZE, dtype)
 
-    def time_sort(self, kind, dtype):
-        np.sort(self.e, kind=kind)
+    def time_sort_inplace(self, kind, dtype, array_type):
+        self.arr.sort(kind=kind)
 
-    def time_sort_random(self, kind, dtype):
-        np.sort(self.o, kind=kind)
+    def time_sort(self, kind, dtype, array_type):
+        np.sort(self.arr, kind=kind)
 
-    def time_sort_inplace(self, kind, dtype):
-        self.e.sort(kind=kind)
+    def time_argsort(self, kind, dtype, array_type):
+        np.argsort(self.arr, kind=kind)
 
-    def time_sort_equal(self, kind, dtype):
-        self.equal.sort(kind=kind)
+    def array_random(self, size, dtype):
+        arr = np.arange(size, dtype=dtype)
+        np.random.shuffle(arr)
+        return arr
+    
+    def array_ordered(self, size, dtype):
+        return np.arange(size, dtype=dtype)
 
-    def time_sort_many_equal(self, kind, dtype):
-        self.many_equal.sort(kind=kind)
+    def array_reversed(self, size, dtype):
+        return np.arange(size-1, -1, -1, dtype=dtype)
 
-    def time_argsort(self, kind, dtype):
-        self.e.argsort(kind=kind)
+    def array_uniform(self, size, dtype):
+        return np.ones(size, dtype=dtype)
 
-    def time_argsort_random(self, kind, dtype):
-        self.o.argsort(kind=kind)
+    def array_sorted_block_10(self, size, dtype):
+        return self.sorted_block(size, dtype, 10)
+
+    def array_sorted_block_100(self, size, dtype):
+        return self.sorted_block(size, dtype, 100)
+
+    def array_sorted_block_1000(self, size, dtype):
+        return self.sorted_block(size, dtype, 1000)
+
+    def array_swapped_pair_1_percent(self, size, dtype):
+        return self.swapped_pair(size, dtype, 0.01)
+
+    def array_swapped_pair_10_percent(self, size, dtype):
+        return self.swapped_pair(size, dtype, 0.1)
+
+    def array_swapped_pair_50_percent(self, size, dtype):
+        return self.swapped_pair(size, dtype, 0.5)
+
+    AREA_SIZE = 10
+
+    def array_random_unsorted_area_50_percent(self, size, dtype):
+        return self.random_unsorted_area(size, dtype, 0.5, self.AREA_SIZE)
+
+    def array_random_unsorted_area_10_percent(self, size, dtype):
+        return self.random_unsorted_area(size, dtype, 0.1, self.AREA_SIZE)
+
+    def array_random_unsorted_area_1_percent(self, size, dtype):
+        return self.random_unsorted_area(size, dtype, 0.01, self.AREA_SIZE)
+
+    BUBBLE_SIZE = 10
```

Comment:
might move this up top with a comment as to what it means.

Reply:
Done!